sandbox.go

```diff
@@ -263,7 +266,45 @@ func (sb *sandbox) updateGateway(ep *endpoint) error {
 	return nil
 }
 
+func (sb *sandbox) SetKey(basePath string) error {
+	var err error
+	if basePath == "" {
+		return types.BadRequestErrorf("invalid sandbox key")
+	}
+
+	sb.Lock()
+	if sb.osSbox != nil {
+		sb.Unlock()
+		return types.ForbiddenErrorf("failed to set sandbox key : already assigned")
+	}
+
+	sb.osSbox, err = osl.GetSandboxForExternalKey(basePath, sb.Key())
```

Comment:
Can we not keep holding the lock when calling `GetSandboxForExternalKey`? Instead we can receive the return variable in a local variable and the hold the lock again before update `sb` with it.